File: Selection_Sort/Selection_Sort.R
n = as.integer(readline())
arr = c()

for (i in 1:n) {
	arr[i] = as.integer(readline())
}

selectionSort<-function(arr){
	for (i in 1:(length(arr)-1)) {
		min = i
		for (j in (i+1):length(arr)) {
			if(arr[j] < arr[min]){
				min = j
			}
		}
		if(min != i){
			arr[c(min,i)] = arr[c(i,min)]
		}
	}

	return(arr);
}

printArray<-function(arr){

	output = "[";
	for (i in 1:length(arr)) {
		output = paste(output," ",arr[i]," ")
		if(i<length(arr)-1)
			output = paste(output,",")
	}
	output = paste(output,"]")

	print(output)
}

arr = selectionSort(arr)
printArray(arr)

File: Selection_Sort/Selection_Sort.php
<?php

function swap(&$arr,$i,$j){
	$temp = $arr[$i];
	$arr[$i] = $arr[$j];
	$arr[$j] = $temp;

	return;
}
function selectionSort(&$arr)
{
	$min = 0;
	for ($i=0; $i < count($arr)-1; $i++) { 
		$min = $i;
		for ($j=$i+1; $j < count($arr); $j++) { 
			if ($arr[$j] < $arr[$min]) {
				$min = $j;
			}
		}
		if ($min != $i) {
			swap($arr,$i,$min);
		}
	}
	return;
}

function printArray($arr){

	$output = "[";
	for ($i=0; $i <count($arr); $i++) { 
		$output .= " ".$arr[$i]." ";
		if ($i<count($arr)-1) {
			$output .=",";
		}
	}
	$output .="]";
	echo $output;

	return;
}

$n = (int)readline();
$arr = array();

for ($i=0; $i < $n; $i++) { 
	$arr[$i] = (int)readline();
}

selectionSort($arr);
printArray($arr);
?>